### DISCUSSION TITLE
STweep CLI

### DISCUSSION THREAD

**Discussion by Barteling** [STweep/STweep#32] [Ideas]
As I mentioned before (#3) I want to make a STweep command line interface to allow STweep integration in build servers and attachments to commit hooks.

With some of you have had already some discussions of how this could look like, but I thought it would be a good idea to make this a public discussion. 

Right now I have this in mind: 
A windows command line executable STweepCLI
Where you can do something like this:
**STweep a file:** 
STweepCLI.exe --File 'C:/Main.TcPou' --SettingsFilePath 'C:/SettingsFile' 

**STweep a folder:** 
STweepCLI.exe --Folder 'C:/Myproject' --SettingsFilePath 'C:/SettingsFile'
STweepCLI.exe --Folder 'C:/Myproject' --SettingsFilePath 'C:/SettingsFile' --Recursive
STweepCLI.exe --Folder 'C:/Myproject' --SettingsFilePath 'C:/SettingsFile' --Recursive  --Include 'SomeFilter' --Exclude 'Somefilter'

Some goals I have:
 - Support TwinCAT files.
 - Support for Codesys files (which for now looks tricky).
 - Support for PLCOpen format.
 - To keep it light weight for build servers I don't want to rely on any IDE. But just act on text files/ folder structure.
 
Let me know your ideas on this topic!

**Comment by Roald87** [comment]
Looks like a good idea. 

Not sure if the recursive is necessary. If you're formatting a project, I assume it will do it recursively. Also the filters might not be needed. Why wouldn't you want a file to look nicely formatted :p. 

There is maybe one thing you could add: a safe mode which builds the project before and after formatting like I do with _TcBlack_. Then you're sure that there are no unintended changes in the code logic. Although this mode is too slow for the commit hooks, it is ok for build servers.

**Comment by nbuchele-KAMP** [comment]
I think this is a great idea. Could really help automate things with my team.
